sympy/solvers/solvers.py

```diff
@@ -375,6 +375,173 @@ def checksol(f, symbol, sol=None, **flags):
+    elif 'ImmutableDenseMatrix' in [type(a).__name__ for a in fi.args]:
+        return fi.lhs - fi.rhs
+    else:
+        return fi.rewrite(Add, evaluate=False)
+
+def _swap_dict(solution, swap_sym):
+    """Helper function to handle pure dict-like solutions like {y(t): 2, x(t): 1/3}"""
+    solution = {swap_sym.get(k, k): v.subs(swap_sym)
+                     for k, v in solution.items()}
+    return solution
+
+def _swap_list(solution, swap_sym):
+    """
+    Helper function to handle list solutions which
+    have an dictionary inside like [{g(x): 0, f(x): 2}]
+    """
+    for i, sol in enumerate(solution):
+        solution[i] = {swap_sym.get(k, k): v.subs(swap_sym)
+                      for k, v in sol.items()}
+    return solution
+
+def _handle_equality(f):
+    if any(_is_eq_bool(fi) for fi in f):
+        for i, fi in enumerate(f):
+            # if not isinstance(fi, (Equality, Unequality)):
+            #     continue
+            args = fi.args
+            if args[1] in (S.true, S.false):
+                args = args[1], args[0]
+            L, R = args
+            if L in (S.false, S.true):
+                if isinstance(fi, Unequality):
+                    L = ~L
+                if R.is_Relational:
+                    pass
+                elif R.is_Symbol:
+                    return L
+                elif R.is_Boolean and (~R).is_Symbol:
+                    return ~L
+                else:
+                    raise NotImplementedError(filldedent('''
+                        Unanticipated argument of Eq when other arg
+                        is True or False.
+                    '''))
+
+def _handle_equality_func(f, as_set):
+#This function handles cases like Eq(x**2, True), Eq(2*x, True) but not Eq(x, True)
+# which has already been handled above
+    f = [fi for fi in f if fi is not S.true]
+    if any(fi is S.false for fi in f):
+        if as_set:
+            return [], set()
+        return []
+
+def _split_real_img(f, bare_f):
+    for i, fi in enumerate(f):
+        # if we can split it into real and imaginary parts then do so
+        # For e.g., if f=solve(x + 3 + y*I), then f will be modified to in this block [x+3, y]
+        # Hence, bare_f is also changed as we get a system of two equations from a single equation
+        freei = f[i].free_symbols
+        if freei and all(s.is_extended_real or s.is_imaginary for s in freei):
+            fr, fi = f[i].as_real_imag()
+            # accept as long as new re, im, arg or atan2 are not introduced
+            had = f[i].atoms(re, im, arg, atan2)
+            if fr and fi and fr != fi and not any(
+                    i.atoms(re, im, arg, atan2) - had for i in (fr, fi)):
+                if bare_f:
+                    bare_f = False
+                f[i: i + 1] = [fr, fi]
+    return f, bare_f
+
+def _handle_abs_args(f, symbols):
+    # this function converts Abs into Piecewise, arg into Piecewise and atan
+    for i, fi in enumerate(f):
```

Comment:
This would be cleaner as a function that operates on single equations rather than a list of equations.